reader.lua

```diff
@@ -206,87 +231,68 @@ if G_reader_settings:isTrue("color_rendering") and not Device:hasColorScreen() t
     })
 end
 
+-- Get which file to start with
+local last_file = G_reader_settings:readSetting("lastfile")
+local start_with = G_reader_settings:readSetting("start_with") or "filemanager"
+
 -- Helpers
-local lfs = require("libs/libkoreader-lfs")
 local function retryLastFile()
     local ConfirmBox = require("ui/widget/confirmbox")
     return ConfirmBox:new{
         text = _("Cannot open last file.\nThis could be because it was deleted or because external storage is still being mounted.\nDo you want to retry?"),
         ok_callback = function()
-            local last_file = G_reader_settings:readSetting("lastfile")
-            if lfs.attributes(last_file, "mode") == "file" then
-                local ReaderUI = require("apps/reader/readerui")
-                UIManager:nextTick(function()
-                    ReaderUI:showReader(last_file)
-                end)
-            else
+            if lfs.attributes(last_file, "mode") ~= "file" then
                 UIManager:show(retryLastFile())
             end
         end,
+        cancel_callback = function()
+            start_with = "filemanager"
+        end,
     }
 end
 
-local function getPathFromURI(str)
-    local hexToChar = function(x)
-        return string.char(tonumber(x, 16))
-    end
-
-    local unescape = function(url)
-       return url:gsub("%%(%x%x)", hexToChar)
-    end
-
-    local prefix = "file://"
-    if str:sub(1, #prefix) ~= prefix then
-        return str
-    end
-    return unescape(str):sub(#prefix+1)
-end
-
--- Get which file to start with
-local last_file = G_reader_settings:readSetting("lastfile")
-local start_with = G_reader_settings:readSetting("start_with")
-local open_last = start_with == "last"
-
-if open_last and last_file and lfs.attributes(last_file, "mode") ~= "file" then
-    UIManager:show(retryLastFile())
-    last_file = nil
+-- Start app
+local exit_code
+if file then
+    local ReaderUI = require("apps/reader/readerui")
+    UIManager:nextTick(function()
+        ReaderUI:showReader(file)
+    end)
+    exit_code = UIManager:run()
+elseif directory then
+    local FileManager = require("apps/filemanager/filemanager")
+    UIManager:nextTick(function()
+        FileManager:setRotationMode(true)
```

Comment:
This one becomes redundant after #7096, right?

Reply:
It will be, I will take care of it there after this is merged, I just want to keep the changes separate